app/models/user.rb

```diff
@@ -1,9 +1,15 @@
+# app/models/user.rb
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id', dependent: :destroy
-  has_many :comments, dependent: :destroy
-  has_many :likes, dependent: :destroy
+  # Validations
+  validates :name, presence: true
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
-  def recent_posts(limit = 3)
-    posts.order(created_at: :desc).limit(limit)
+  # Associations
+  has_many :posts, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  # Method to retrieve the 3 most recent posts for a user
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
   end
 end
```

Comment:
- [ ] Nice job you've done 👏 . However, you're tasked to add unit specs for all validations in your models as well.

### Instructions:
![image](https://github.com/massazoum/Rails_App/assets/50990925/8dc17759-8c8f-4718-9b47-50692f3335fa)

- [ ] Kindly do well to do so.

### Sample validation spec code:
```
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```